### PR TITLE
Respect previous LD_LIBRARY_PATH

### PR DESCRIPTION
**Pull request by Saviq**

### CODE REVIEW

File: env-setup/hacks/setup-gl [hunk 1]
```diff
@@ -3,4 +3,4 @@ set -x
 
 export __EGL_VENDOR_LIBRARY_DIRS=$SNAP/etc/glvnd/egl_vendor.d:$SNAP/usr/share/glvnd/egl_vendor.d
 export LIBGL_DRIVERS_PATH=$SNAP/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/dri
-export LD_LIBRARY_PATH=${SNAP}/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}:${SNAP}/lib/${SNAPCRAFT_ARCH_TRIPLET}
+export LD_LIBRARY_PATH=${SNAP}/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}:${SNAP}/lib/${SNAPCRAFT_ARCH_TRIPLET}:$LD_LIBRARY_PATH
```
Comment (AlanGriffiths) [inline env-setup/hacks/setup-gl:6]: Need to check that `LD_LIBRARY_PATH` is set before appending it.

Comment (Saviq) [reply env-setup/hacks/setup-gl:6]: Not really, a trailing colon doesn't break anything… But can do if you say so.

Comment (AlanGriffiths) [reply env-setup/hacks/setup-gl:6]: Somewhere I saw something like:

```
export LD_LIBRARY_PATH=/foo/bar:${LD_LIBRARY_PATH}
```
is a security risk, you should be using:
```
export LD_LIBRARY_PATH=/foo/bar${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
```

Comment (AlanGriffiths) [reply env-setup/hacks/setup-gl:6]: But I never tried to understand the logic

Comment (Saviq) [reply env-setup/hacks/setup-gl]: Fixed.

**Review by AlanGriffiths — APPROVED** [review]
OK

bors merge

**Comment by AlanGriffiths** [comment]
@Saviq have you noticed the Github action fails here?

**Comment by Saviq** [comment]
> @Saviq have you noticed the Github action fails here?

Yes, it's a race in the action. If it happens to often, I'll modify to use a different action to setup lxd.

**Pull request merged**